### PR TITLE
Update Readme: Add note re: support for the HTTP gem

### PR DESCRIPTION
**Pull request by BKSpurgeon**
Folks when reviewing will be able to quickly see what clients are supported.

### CODE REVIEW

**Comment by bblimke** [comment]
@BKSpurgeon HTTP is already listed 3 lines above.

**Comment by BKSpurgeon** [comment]
> @BKSpurgeon HTTP is already listed 3 lines above.

You are right! 

I completely missed it! I figured that the real problem is that it is not listed alphabetically; and luckily for me through fortuitous accident, I could in fact use it!

I have updated.

Thank you for reviewing.

File: README.md [hunk 1]
```diff
@@ -24,16 +24,19 @@ Features
...
-* Patron
-* EM-HTTP-Request
+* Async::HTTP::Client
 * Curb (currently only Curb::Easy)
-* Typhoeus (currently only Typhoeus::Hydra)
+* EM-HTTP-Request
 * Excon
-* HTTP Gem
+* HTTPClient
+* [HTTP (The Gem! also known as http.rb)](https://github.com/httprb/http)
```
Comment (bblimke) [inline README.md:32]: @BKSpurgeon not sure `The Gem!` is necessary?

File: README.md [hunk 2]
```diff
@@ -24,16 +24,19 @@ Features
...
 * Manticore
-* Async::HTTP::Client
+* Net::HTTP and libraries based on Net::HTTP:
```
Comment (bblimke) [inline README.md:34]: @BKSpurgeon WebMock supports these and other clients based on Net::HTTP. This change suggests it only supports the listed 3 libs. I suggest adding `e.g.` or `like`

**Comment by bblimke** [comment]
@BKSpurgeon awesome, thank you!

**Pull request merged**